### ISSUE TITLE
Why does a coredump when I call "AVBufferSinkParams.pixel_fmts().put(...)"

### ISSUE THREAD

**Issue by wjl327** [bytedeco/javacpp-presets#837]
Method code:

 	    AVFilter buffersrc = avfilter_get_by_name("buffer");
        AVFilter buffersink = avfilter_get_by_name("buffersink");
        AVFilterInOut outputs = avfilter_inout_alloc();
        AVFilterInOut inputs  = avfilter_inout_alloc();

        /* buffer video source: the decoded frames from the decoder will be inserted here. */
        AVFilterGraph filter_graph = avfilter_graph_alloc();
        String args = String.format("video_size=%dx%d:pix_fmt=%d:time_base=%d/%d:pixel_aspect=%d/%d",
                avCodecCtx.width(), avCodecCtx.height(), avCodecCtx.pix_fmt(),
                avCodecCtx.time_base().num(), avCodecCtx.time_base().den(),
                avCodecCtx.sample_aspect_ratio().num(), avCodecCtx.sample_aspect_ratio().den());
        AVFilterContext buffersrc_ctx = new AVFilterContext();
        if(avfilter_graph_create_filter(buffersrc_ctx, buffersrc, "in", args, null, filter_graph) < 0) {
            throw new RuntimeException("Cannot create buffer source");
        }

        avfilter.AVBufferSinkParams params = av_buffersink_params_alloc();
        params.pixel_fmts().put(AV_PIX_FMT_YUV420P, AV_PIX_FMT_NONE);

coredump line：
params.pixel_fmts().put(AV_PIX_FMT_YUV420P, AV_PIX_FMT_NONE);  

------------------------------------
Stack: [0x0000700000e70000,0x0000700000f70000],  sp=0x0000700000f6f720,  free space=1021k
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
C  [libjniavutil.dylib+0x3cc4d]  Java_org_bytedeco_javacpp_IntPointer_put__JI+0x4d
j  org.bytedeco.javacpp.IntPointer.put(JI)Lorg/bytedeco/javacpp/IntPointer;+0
j  com.xxx.avfilter.Avffilter.initAvfilter()V+189
j  com.xxx.avfilter.Avffilter.avfilter()V+55
j  com.xxx.avfilter.Bootstrap.main([Ljava/lang/String;)V+18
v  ~StubRoutines::call_stub
V  [libjvm.dylib+0x2f0bea]  JavaCalls::call_helper(JavaValue*, methodHandle*, JavaCallArguments*, Thread*)+0x6ae
V  [libjvm.dylib+0x327b91]  jni_invoke_static(JNIEnv_*, JavaValue*, _jobject*, JNICallType, _jmethodID*, JNI_ArgumentPusher*, Thread*)+0x1bf
V  [libjvm.dylib+0x32091d]  jni_CallStaticVoidMethod+0x15d
C  [java+0x38c6]  JavaMain+0x9c4
C  [libsystem_pthread.dylib+0x3339]  _pthread_body+0x7e
C  [libsystem_pthread.dylib+0x62a7]  _pthread_start+0x46
C  [libsystem_pthread.dylib+0x2445]  thread_start+0xd
C  0x0000000000000000

Java frames: (J=compiled Java code, j=interpreted, Vv=VM code)
j  org.bytedeco.javacpp.IntPointer.put(JI)Lorg/bytedeco/javacpp/IntPointer;+0
j  com.xxx.avfilter.Avffilter.initAvfilter()V+189
j  com.xxx.avfilter.Avffilter.avfilter()V+55
j  com.xxx.avfilter.Bootstrap.main([Ljava/lang/String;)V+18
v  ~StubRoutines::call_stub

-----------------------------------------------
GC Heap History (2 events):
Event: 1.008 GC heap before
{Heap before GC invocations=1 (full 0):
 PSYoungGen      total 76288K, used 65536K [0x000000076ab00000, 0x0000000770000000, 0x00000007c0000000)
  eden space 65536K, 100% used [0x000000076ab00000,0x000000076eb00000,0x000000076eb00000)
  from space 10752K, 0% used [0x000000076f580000,0x000000076f580000,0x0000000770000000)
  to   space 10752K, 0% used [0x000000076eb00000,0x000000076eb00000,0x000000076f580000)
 ParOldGen       total 175104K, used 0K [0x00000006c0000000, 0x00000006cab00000, 0x000000076ab00000)
  object space 175104K, 0% used [0x00000006c0000000,0x00000006c0000000,0x00000006cab00000)
 Metaspace       used 6091K, capacity 6262K, committed 6400K, reserved 1056768K
  class space    used 682K, capacity 722K, committed 768K, reserved 1048576K
Event: 1.013 GC heap after
Heap after GC invocations=1 (full 0):
 PSYoungGen      total 76288K, used 2575K [0x000000076ab00000, 0x0000000770000000, 0x00000007c0000000)
  eden space 65536K, 0% used [0x000000076ab00000,0x000000076ab00000,0x000000076eb00000)
  from space 10752K, 23% used [0x000000076eb00000,0x000000076ed83c70,0x000000076f580000)
  to   space 10752K, 0% used [0x000000076f580000,0x000000076f580000,0x0000000770000000)
 ParOldGen       total 175104K, used 16K [0x00000006c0000000, 0x00000006cab00000, 0x000000076ab00000)
  object space 175104K, 0% used [0x00000006c0000000,0x00000006c0004000,0x00000006cab00000)
 Metaspace       used 6091K, capacity 6262K, committed 6400K, reserved 1056768K
  class space    used 682K, capacity 722K, committed 768K, reserved 1048576K
}

**Comment by saudet** [comment]
You'll need to set `pixel_fmts()` to something first.

**Comment by wjl327** [comment]
init `pixel_fmts()` ?

avfilter.AVBufferSinkParams params = av_buffersink_params_alloc(); 
but not found witch method of `avfilter.AVBufferSinkParams`  is used to initialize `pixel_fmts()` ?

`
//        avfilter.AVBufferSinkParams params = av_buffersink_params_alloc();
//        IntPointer intPointer = new IntPointer(2);
//        intPointer.put(AV_PIX_FMT_YUV420P, AV_PIX_FMT_NONE);
//        params.pixel_fmts().put(intPointer);

//        IntPointer intPointer = new IntPointer(2);
//        avfilter.AVBufferSinkParams params = new avfilter.AVBufferSinkParams(intPointer);
//        params.pixel_fmts().put(AV_PIX_FMT_YUV420P, AV_PIX_FMT_NONE);
`
All the above have failed. Could you tell me how to set, please? thx.

**Comment by saudet** [comment]
There's an example of that here:
https://github.com/FFmpeg/FFmpeg/blob/master/doc/examples/filtering_video.c
If we do the same thing in Java, it should work. Let me know

**Comment by wjl327** [comment]
Yes, it works fine. I found that avfilter.AVBufferSinkParams is optional.

For example :
ret = avfilter_graph_create_filter(&buffersink_ctx, buffersink, "out", NULL, NULL, filter_graph);
instead of 
ret = avfilter_graph_create_filter(&buffersink_ctx, buffersink, "out", NULL, param, filter_graph);

If you want to use avfilter.AVBufferSinkParams ，you may have the above problem.
